cpp/contact_kernels.hpp

```diff
@@ -237,43 +239,23 @@ kernel_fn generate_contact_kernel(
       double lmbda = 0;
       for (int j = offsets[2]; j < offsets[3]; j++)
         lmbda += c[j + c_offset] * phi_coeffs(facet_index, q, j);
-      double gap = 0;
-      int gap_offset = c_offset + offsets[4];
+
       // if normal not constant, get surface normal at current quadrature point
+      int normal_offset = c_offset + offsets[5];
       if (!constant_normal)
       {
         n_dot = 0;
         for (int i = 0; i < gdim; i++)
         {
-          gap += c[gap_offset + q * gdim + i] * c[gap_offset + q * gdim + i];
-          n_surf(i) = c[gap_offset + q * gdim + i];
+          n_surf(i) = -c[normal_offset + q * gdim + i];
```

Comment:
Would be good to document why we use a minus here (and probably every other place we compute n_surf)